### ISSUE TITLE
Use control component in widget sub component

### ISSUE THREAD

**Issue by jonaswjs** [LaxarJS/laxar-vue-adapter#9]
To separate a big widget in multiple components it would be great if we can use the controls of the widget. 

When using a `test-control` in a sub component of a widget, which has the `test-control` listed in the `widget.json` I get the following warning:
```
[Vue warn]: Unknown custom element: <test-control> - did you register the component correctly? For recursive components, make sure to provide the "name" option.
```

**Comment by x1B** [comment]
@jonaswjs I agree that this feature would be very useful. However, I am not sure if/how this can be done with VueJS.

For now, I think that there are two options:

 - Put everything into the widget as sub-components — of course, then you won't be able to use a control from multiple widgets.

 - Or, package all sub-components as controls (then, you can declare the sub-controls from within the controls.json). You might prefix their names with the widget name to indicate the association.

Alternatively, you might be able to register controls *globally* from within the widget's `create` method. Of course, then they'll be available *anywhere*.

It might also be possible to pass controls from a widget to a sub-component by passing the control itself as a *prop* and loading it through an [async wrapper component](https://vuejs.org/v2/guide/components.html#Async-Components) into the child, but we have not tried this yet.
